File: branches/Medusa/Source/Fantasy.Jobs/JobProperty.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.XSerialization;
using System.Collections;
using System.Xml.Linq;

namespace Fantasy.Jobs
{
    [XSerializable("property", NamespaceUri=Consts.XNamespaceURI)]
    [Serializable]
    public class JobProperty: IXSerializable, ICloneable , IConditionalObject  
    {
        public string Name { get; set; }

        [XValue]
        public string Value { get; set; }

        public object Clone()
        {
            return new JobProperty { Name = this.Name, Value = this.Value, Condition = this.Condition }; 
        }

        public void Load(IServiceProvider context, XElement element)
        {
            this.Name = element.Name.LocalName;
            XHelper.Default.LoadByXAttributes(context, element, this);
        }

        public void Save(IServiceProvider context, XElement element)
        {
            XHelper.Default.SaveByXAttributes(context, element, this);
        }

        [XAttribute("condition")] 
        public string Condition { get; set; }
       
    }


    class JobPropertiesSerializer : IXCollectionSerializer
    {

        public void Save(IServiceProvider context, XElement element, IEnumerable collection)
        {
            

            foreach (JobProperty item in collection)
            {
                XElement childElement = new XElement(element.Name.Namespace + item.Name);
                element.Add(childElement);
                item.Save(context, childElement);
            }
        }

        public IEnumerable Load(IServiceProvider context, XElement element)
        {
            ArrayList rs = new ArrayList();
            foreach (XElement childElement in element.Elements())
            {
                JobProperty item = new JobProperty();
                item.Load(context, childElement);
                rs.Add(item);
            }

            return rs;
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Jobs/JobProperty.java
package Fantasy.Jobs;

import Fantasy.XSerialization.*;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[XSerializable("property", NamespaceUri=Consts.XNamespaceURI)]
public class JobProperty extends IXSerializable implements Cloneable, IConditionalObject, Serializable
{
	private String privateName;
	public final String getName()
	{
		return privateName;
	}
	public final void setName(String value)
	{
		privateName = value;
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XValue]
	private String privateValue;
	public final String getValue()
	{
		return privateValue;
	}
	public final void setValue(String value)
	{
		privateValue = value;
	}

	public final Object clone()
	{
		JobProperty tempVar = new JobProperty();
		tempVar.setName(this.getName());
		tempVar.setValue(this.getValue());
		tempVar.setCondition(this.getCondition());
		return tempVar;
	}

	public final void Load(IServiceProvider context, XElement element)
	{
		this.setName(element.getName().LocalName);
		XHelper.Default.LoadByXAttributes(context, element, this);
	}

	public final void Save(IServiceProvider context, XElement element)
	{
		XHelper.Default.SaveByXAttributes(context, element, this);
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XAttribute("condition")]
	private String privateCondition;
	public final String getCondition()
	{
		return privateCondition;
	}
	public final void setCondition(String value)
	{
		privateCondition = value;
	}

}